### ISSUE TITLE
datadog-plugin-graphql is not validating if directives is an array

### ISSUE THREAD

**Issue by mabilbao** [DataDog/dd-trace-js#4097]
Hey guys,

After this [PR](https://github.com/DataDog/dd-trace-js/pull/3819/), I started having an issue with this [line](https://github.com/DataDog/dd-trace-js/pull/3819/files/8f427dca6bbc2d58544a7e8247fc38ba35ba2323#diff-796b4c9573e0971daf318484cdd9517eb62524e5dc46764d8342adaa5339c29aR126) (mentioned in a late [comment](https://github.com/DataDog/dd-trace-js/pull/3819#discussion_r1499965660) by myself also).
```
  const directives = info.fieldNodes[0].directives
  for (const directive of directives) {
```

---

I am using a product called: KesytoneJS and the schema that it generates does not always return directives. For example:
```
🚀 ~ getResolverInfo ~ info.fieldNodes[0]: {
  kind: 'Field',
  name: { kind: 'Name', value: 'user' },
  arguments: [ { kind: 'Argument', name: [Object], value: [Object] } ],
  selectionSet: {
    kind: 'SelectionSet',
    selections: [ [Object], [Object], [Object] ],
    loc: Location {
      start: 19,
      end: 108,
      startToken: [Token],
      endToken: [Token],
      source: [Source]
    }
  }
}
```

Finally, the error I am getting is: `TypeError: directives is not iterable`. 

I keep my project working with version `4.21.0` that doesn't have that change, but now I have a high dependabot issue due to that version is using `lodash.pick`. 
It looks like Lodash doesn't have a patch version yet: https://github.com/advisories/GHSA-p6mc-m468-83gw.
It looks like Datadog just removed the dependency in new versions: 
* https://github.com/DataDog/dd-trace-js/releases/tag/v5.2.0
* https://github.com/DataDog/dd-trace-js/releases/tag/v4.26.0

---

Should we validate if `directives` is effectively iterable before the `for`?

I've tried to push a fix for this, but I am getting 403 when I push 🥲 I think we should do something like:
```ts
  const directives = info.fieldNodes[0].directives
  if (Array.isArray(directives)) {
    for (const directive of directives) {
      const argList = {}
      for (const argument of directive['arguments']) {
        argList[argument.name.value] = argument.value.value
      }

      if (Object.keys(argList).length) {
        resolverVars[directive.name.value] = argList
      }
    }
  }
```

Thanks in advance!

**Comment by simon-id** [comment]
Once again, thank you for reporting this issue with detailed infos. We'll look into it asap and update the thread when we have news. Although this issue won't be resolved until next week as it's already friday afternoon for us :/

**Comment by mabilbao** [comment]
> Once again, thank you for reporting this issue with detailed infos. We'll look into it asap and update the thread when we have news. Although this issue won't be resolved until next week as it's already friday afternoon for us :/

Thank you @simon-id! No rush at all 😄

**Comment by uurien** [comment]
PR with the fix has been merged: https://github.com/DataDog/dd-trace-js/pull/4116
It's going to be fixed in the next release.
thanks @mabilbao

**Comment by mabilbao** [comment]
> PR with the fix has been merged: #4116 It's going to be fixed in the next release. thanks @mabilbao

Thank you so much @uurien! This was super fast 😄

**Comment by uurien** [comment]
Hello,
Last release should fix the problem ([v5.6.0](https://github.com/DataDog/dd-trace-js/releases/tag/v5.6.0), [v4.30.0](https://github.com/DataDog/dd-trace-js/releases/tag/v4.30.0), [3.51.0](https://github.com/DataDog/dd-trace-js/releases/tag/v3.51.0)).

I'm closing the issue.
Thanks @mabilbao
